canal/dump.go

```diff
@@ -70,6 +70,8 @@ func (h *dumpParseHandler) Data(db string, table string, values []string) error
 	}
 
 	events := newRowsEvent(tableInfo, InsertAction, [][]interface{}{vs})
+	nextPos := h.c.syncer.GetNextPosition()
+	events.NextPos = nextPos
```

Comment:
this NextPos is not the next pos of current event. You can use the event header Pos directly.

Reply:
where is the event header pos? like this?
```
nextPos := mysql.Position{h.name, h.pos}
events.NextPos = nextPos
```

Reply:
``
nextPos := mysql.Position{h.name, uint32(h.pos)}
```

Reply:
This is for dump, and I think `syncer.GetNextPosition` can get the correct position here. You can only get the binlog sync pos after dump.

Reply:
I'm so sorry I can't understand your opinion, which is right? 

nextPos := h.c.syncer.GetNextPosition()
or 
nextPos := mysql.Position{h.name, uint32(h.pos)}

if I add a param in "newRowsEvent(t, action, ev.Rows)",  should modify here too.

Reply:
both are not right here, here in the dump produce, we can't know anything about the sync position. So you can use the position here.